File: front/src/app/models/producto.ts
import type { Imagen } from './imagen';

export interface Producto {
  id: number;
  nombre: string;
  descripcion: string;
  precio: number;
  stock: number;
  categoria_id: number;
  activo: boolean;
  imagenes: Imagen[];
  creado_en: string;
  actualizado_en: string;
}

export interface ProductoInput {
  nombre: string;
  descripcion: string;
  precio: number;
  stock: number;
  categoria_id: number;
  activo: boolean;
}

export interface PaginatedProductos {
  data: Producto[];
  total: number;
  offset: number;
  limit: number;
}

File: back/domain/inputs/producto.go
package inputs

type ProductoInput struct {
	Nombre      string  `json:"nombre"`
	Descripcion string  `json:"descripcion"`
	Precio      float64 `json:"precio"`
	Stock       int     `json:"stock"`
	CategoriaID int64   `json:"categoria_id"`
	Activo      bool    `json:"activo"`
}

type ProductoUpdateInput struct {
	Nombre      *string  `json:"nombre,omitempty"`
	Descripcion *string  `json:"descripcion,omitempty"`
	Precio      *float64 `json:"precio,omitempty"`
	Stock       *int     `json:"stock,omitempty"`
	CategoriaID *int64   `json:"categoria_id,omitempty"`
	Activo      *bool    `json:"activo,omitempty"`
}
